packages/cozy-dataproxy-lib/src/search/SearchEngine.ts

```diff
@@ -197,25 +208,83 @@ export class SearchEngine {
     this.subscribeDoctype(this.client, CONTACTS_DOCTYPE)
     this.subscribeDoctype(this.client, APPS_DOCTYPE)
 
+    const pouchLink = getPouchLink(this.client)
+    if (pouchLink) {
+      const sharedDrivesDoctypes = pouchLink.getSharedDriveDoctypes()
+      for (const sharedDrivesDoctype of sharedDrivesDoctypes) {
+        const driveId = sharedDrivesDoctype.split('-').pop()
+        if (driveId) {
+          this.addSharedDriveRealtime(driveId)
+        }
+      }
+    }
+
     if (this.isLocalSearch) {
       this.debouncedReplication()
     }
     // The document indexing should be performed once everything is setup
     await this.indexDocumentsAtInit()
   }
 
-  subscribeDoctype(client: CozyClient, doctype: string): void {
+  addSharedDrive(driveId: string): void {
+    this.addSharedDriveRealtime(driveId)
+    if (this.isLocalSearch) {
+      this.debouncedReplication()
+    }
+  }
+
+  removeSharedDrive(driveId: string): void {
+    this.sharedDrivesRealtimes[driveId]?.stop()
+    delete this.sharedDrivesRealtimes[driveId]
+    if (
+      this.searchIndexes &&
+      this.searchIndexes[`${SHARED_DRIVE_FILES_DOCTYPE}-${driveId}`]
+    ) {
+      delete this.searchIndexes[`${SHARED_DRIVE_FILES_DOCTYPE}-${driveId}`]
+    }
+    if (this.isLocalSearch) {
+      this.debouncedReplication()
```

Comment:
I'm not sure it's useful to start the replication here?

Reply:
Right, no reason to launch a replication here since the index is removed, it won't appear in the search results anyway